src/libsystemd/sd-journal/journal-def.h

```diff
@@ -102,18 +102,12 @@ assert_cc(sizeof(struct FieldObject) == sizeof(struct FieldObject__packed));
         le64_t xor_hash;                               \
         union {                                        \
                 struct {                               \
-                        dummy_t __empty__regular;      \
-                        struct {                       \
-                                le64_t object_offset;  \
-                                le64_t hash;           \
-                        } regular[];                   \
-                };                                     \
+                        le64_t object_offset;          \
+                        le64_t hash;                   \
+                } regular[0];                          \
                 struct {                               \
-                        dummy_t __empty_compact;       \
-                        struct {                       \
-                                le32_t object_offset;  \
-                        } compact[];                   \
-                };                                     \
+                        le32_t object_offset;          \
+                } compact[0];                          \
         } items;                                       \
```

Comment:
so here you went for the [0] approach, which suggests the payload[0] in the previous commit should also be fine, no?

Reply:
kinda would prefer if there was a comment here, that explains briefly what the point of the [0] is. i.e. something like `/* this is an array; since we are not allowed to place a variable sized array in a union, we just zero-size it, even if it is generally longer */`